src/aggregate/aggregate_exec.c

```diff
@@ -351,14 +353,31 @@ static RedisModuleCtx *blockedClientReqCtx_getRedisctx(const blockedClientReqCtx
 }
 
 static void blockedClientReqCtx_destroy(blockedClientReqCtx *BCRctx) {
+  if (BCRctx->req) {
+    AREQ_Free(BCRctx->req);
+  }
+  RS_CHECK_FUNC(RedisModule_BlockedClientMeasureTimeEnd, BCRctx->blockedClient);
   RedisModule_UnblockClient(BCRctx->blockedClient, NULL);
+  WeakRef_Release(BCRctx->spec_ref);
   rm_free(BCRctx);
 }
 
 void AREQ_Execute_Callback(blockedClientReqCtx *BCRctx) {
   RedisModuleCtx *redis_ctx = BCRctx->req->sctx->redisCtx = RedisModule_GetThreadSafeContext(BCRctx->blockedClient);
   BCRctx->req->reqflags |= QEXEC_F_HAS_THCTX;
-  // lockspec
+
+  StrongRef execution_ref = WeakRef_Promote(BCRctx->spec_ref);
+  if (!StrongRef_Get(execution_ref)) {
+    // The index was dropped while the query was in the job queue.
+    // Notify the client that the query was aborted
+    QueryError status = {0};
+    QueryError_SetError(&status, QUERY_ENOINDEX, "The index was dropped before the query could be executed");
+    QueryError_ReplyAndClear(redis_ctx, &status);
+    blockedClientReqCtx_destroy(BCRctx);
+    return;
+  }
+
+  // lock spec
   RedisSearchCtx_LockSpecRead(BCRctx->req->sctx);
```

Comment:
Maybe we can create the search ctx here after we know the spec is valid?

Reply:
It's created (allocated) before going to the BG. We update its `RedisModuleCtx` first to be linked to the blocked client so we can reply to him if the spec is invalid